gse-demo/src/main/java/com/powsybl/gse/demo/GseDemo.java

```diff
@@ -9,18 +9,102 @@
+        //if we are missing jars or if the magic javalauncher has a problem.
+        //Try to show the error everywhere so that the user has the best chance of noticing it...
+        try {
+            //Try stdout
+            System.out.println(UNCAUGHT_EXCEPTION_MESSAGE);
+            t.printStackTrace();
+        } catch (Throwable throwableSystemOut) {
+            //Swallow throwableSystemOut here, we don't want to confuse the user.
+            //This is such a broken case that we don't want to
+            //log throwableSystemOut, we want the other mechanisms in this method
+            //to log the rootThrowable to fix the root cause.
+        }
+
+        try {
+            //Try a file in the app directory, next to powsybl.log
+            PrintWriter printWriterApp = new PrintWriter(STARTUP_ERROR_LOGFILE_NAME);
+            t.printStackTrace(printWriterApp);
+            printWriterApp.flush();
+        } catch (Throwable throwableFileApp) {
+            //Swallow throwableFileApp here, we don't want to confuse the user.
+            //This is such a broken case that we don't want to
+            //log throwableFileApp, we want the other mechanisms in this method
+            //to log the rootThrowable to fix the root cause.
+        }
+        try {
+            //Try a file in the root directory, next to the executable
+            //The name in alphabetical order should be next to the executable.
+            PrintWriter printWriterRoot = new PrintWriter("../" + STARTUP_ERROR_LOGFILE_NAME);
+            t.printStackTrace(printWriterRoot);
+            printWriterRoot.flush();
+        } catch (Throwable throwableFileRoot) {
+            //Swallow throwableFileRoot here, we don't want to confuse the user.
+            //This is such a broken case that we don't want to
+            //log throwableFileRoot, we want the other mechanisms in this method
+            //to log the rootThrowable to fix the root cause.
+        }
+    }
+
+    //Using javapackager from jdk8,
+    //we must avoid throwing an exception at all costs because
+    //the javapackager launcher just prints an InvocationException
+    //message on stdout. This is the message you get, it has no information:
+    //  GseDemo Error invoking method.
+    //  GseDemo Failed to launch JVM
+    //
+    //Also stdout is not always available, most notably when a user
+    //launches the application by double clicking in a file explorer
     public static void main(String[] args) {
-        SLF4JBridgeHandler.removeHandlersForRootLogger();
-        SLF4JBridgeHandler.install();
+        //Don't use a static logger so that this method can run
+        //even if there is a problem loading the logger classes
+        Logger logger = null;
+        try {
+            logger = LoggerFactory.getLogger(GseDemo.class);
+
+            SLF4JBridgeHandler.removeHandlersForRootLogger();
+            SLF4JBridgeHandler.install();
```

Comment:
removeHandlersForRootLogger and install deal with redirecting Jul to slf4j, so it's independant from getting a logger. If for some reason removeHandlersForRootLogger or install throw, then by having created the logger first, we get nicer logs (using slf4j).

Yes I'm trying to log errors in every case, because the caller (java8 packager) swallows errors.